tests/10-vcdm2.js

```diff
@@ -229,70 +220,58 @@ describe('Types', function() {
+        }
+      );
       // FIXME this needs to be expanded into at least 6 different tests
       // Verifiable Credential MUST have a type specified
       // Verifiable Presentation MUST have a type specified
-      // Proof MUST have a type specified.
       // credentialStatus MUST have a type specified.
       // termsOfUse MUST have a type specified.
       // evidence MUST have a type specified.
-      it('list: "objects that MUST have a type specified."',
-        async function() {
-          this.test.link = `https://w3c.github.io/vc-data-model/#types:~:text=the%20following%20table%20lists%20the%20objects%20that%20MUST%20have%20a%20type%20specified.`;
-          // (Verifiable) credential requires type VerifiableCredential
-          // (Verifiable) presentation requires type VerifiablePresentation
-          // Additional (more specific) types for these are optional.
-          // Missing type property is tested separately.
-          await endpoints.issue(require(
-            './input/credential-optional-type-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-missing-required-type-fail.json')));
-          const presentationOptionalType = await endpoints.createVp({
-            presentation: require('./input/presentation-optional-type-ok.json'),
-            options: createOptions
-          });
-          await endpoints.verifyVp(
-            presentationOptionalType,
-            verifyPresentationOptions
-          );
-          await assert.rejects(
-            endpoints.verifyVp(require(
-              './input/presentation-missing-required-type-fail.json')));
-          // Other objects requiring type: proof, credentialStatus, termsOfUse,
-          // and evidence.
-          // Note: testing proof requires the issuer to allow the input
-          // credential to have an existing proof property.
-          await endpoints.issue(require('./input/credential-proof-ok.json'));
-          await assert.rejects(endpoints.verify(require(
-            './input/credential-proof-missing-type-fail.json')));
-          await endpoints.issue(require('./input/credential-status-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-status-missing-type-fail.json')));
-          await endpoints.issue(require(
-            './input/credential-termsofuse-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-termsofuse-missing-type-fail.json')));
-          await endpoints.issue(require('./input/credential-evidence-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-evidence-missing-type-fail.json')));
+      it('VerifiableCredential, VerifiablePresentation, credentialStatus, ' +
+        'termsOfUse, and evidence are "objects that MUST have a type ' +
+        'specified."', async function() {
```

Comment:
This list should also contain [`refreshService`](https://w3c.github.io/vc-data-model/#refreshing:~:text=Each%20refreshService%20value%20MUST%20specify%20its%20type.)

Reply:
Same for [`credentialSchema`](https://w3c.github.io/vc-data-model/#data-schemas:~:text=Each%20credentialSchema%20MUST%20specify%20its%20type%20(for%20example%2C%20JsonSchema)

Reply:
I think this is what @PatStLouis suggests...
```suggestion
      it('VerifiableCredential, VerifiablePresentation, credentialStatus, ' +
        'termsOfUse, refreshService, credentialSchema, and evidence are ' +
        '"objects that MUST have a type specified."', async function() {
```

Reply:
Then that's a spec change that needs making, as the quote is from this section which references a table which lacks `refreshService` and `credentialSchema`: https://w3c.github.io/vc-data-model/#refreshing:~:text=objects%20that%20MUST%20have%20a%20type

I'll submit a PR there...and then we can update this test once that's merged into the spec.

Reply:
@TallTed @PatStLouis submitted https://github.com/w3c/vc-data-model/pull/1530